app/models/user.server.ts

```diff
@@ -17,7 +18,7 @@ export async function getUserByUsername(username: PrismaUser['username']) {
   }
 
   // Decorate with the user's base domain as well
-  return { ...user, baseDomain: buildUserBaseDomain(username) };
+  return { ...user, baseDomain: buildUserBaseDomain(username), isAdmin: isAdmin(username) };
```

Comment:
Throw the `effectiveUsername` on here too.  So we know: 

- are you an admin?
- what is your real username? (who are you really)
- what is your current effective username? (which user are you actually pretending to be)

Reply:
I'm not sure how to do this since my effective user name is currently passed as a cookie from request to request on the header and we don't have access to the header in the getUserByUsername method.